### DISCUSSION TITLE
Offer completions when overriding class methods

### DISCUSSION THREAD

**Discussion by luabud** [microsoft/pylance-release#2950] [Enhancement]
It would be nice if we could offer completions when users start typing override

**Comment by erictraut** [comment]
Pylance already offers a form of completions for overrides. Do you have something different / more in mind?

![Screen Shot 2022-06-17 at 12 38 22 PM](https://user-images.githubusercontent.com/7040122/174391052-17ae889c-c5d5-4781-bbaa-e15d11c86612.png)

After accepting the completion:

![Screen Shot 2022-06-17 at 12 38 32 PM](https://user-images.githubusercontent.com/7040122/174391059-e5c9e449-86f3-4774-99b8-61db55190edd.png)

**Reply by PythonCoderAS** [reply]
This requires you to know at least part of the name to override.

**Comment by judej** [comment]
+1
[Get a list of methods for override in subclass#3842](https://github.com/microsoft/pylance-release/issues/3842)

**Comment by slcnx** [comment]
@erictraut 
Hello, in your case, the parent class and the subclass need to be in the same file, so that the method of the parent class can be easily overridden in the subclass.
Suppose there are multiple methods in the parent class, in the A file. In file B, I need to inherit the parent class and rewrite some methods. How can I rewrite the methods conveniently and quickly without knowing in advance what methods the parent class has?
Pycharm provides shortcut keys for ctrl + o and override method, which can get the list of parent classes, subclasses and their methods at once.
I want to know, is there any way in vscode to quickly override the method of the parent class?

**Reply by slcnx** [reply]
@judej
